### PR TITLE
Query builder is refactored to package model

### PR DESCRIPTION
**Pull request by ganeshdogiparthi-db**
1. Removed the query config up on a suggestion from Sundar
2. Moved most of the functions moved to data class or base query builder.
3. All other query builders extend the base query builder.

### CODE REVIEW

File: src/databricks/labs/remorph/reconcile/query_builder/hash.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+from io import StringIO
```
Comment (nfx) [inline src/databricks/labs/remorph/reconcile/query_builder/hash.py:1]: This filename overlaps with hash() builtin function. Rename to avoid problems

**Review by nfx — CHANGES_REQUESTED** [review]
Mostly good, but rename the hash file not to collide and fix formatting

File: src/databricks/labs/remorph/reconcile/query_builder/hash.py (hunk 1 above)
Comment (ganeshdogiparthi-db) [reply src/databricks/labs/remorph/reconcile/query_builder/hash.py]: Done. Changed the file name to hash_query and test case respectively. I fixed most of the fmt issues. But still giving some weird fmt issues.

Comment (ganeshdogiparthi-db) [reply src/databricks/labs/remorph/reconcile/query_builder/hash.py]: Made structural changes up on suggestion by Sundar

**Review by nfx — APPROVED** [review]

**Review by sundarshankar89 — APPROVED** [review]
LGTM

**Pull request merged**